cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_splitter_impl.hpp

```diff
@@ -66,16 +67,34 @@ class train_best_split_impl {
         Index node_count,
         const bk::event_vector& deps = {});
 
-    static sycl::event compute_best_split_single_pass(
+    /// Computing random split for `node_count` nodes using `selected_ftr_list`
+    /// and `random_bins_com` values for splitting. Computes best split among randomly
+    /// selected tresholds for each node.
+    ///
+    /// @param[in] queue                    sycl parallel queue
+    /// @param[in] ctx                      a train context structure for GPU backend
+    /// @param[in] node_hist_list           a pre-calculated histogram for each node
+    /// @param[in] selected_ftr_list        a subset of feature indices selected for each node
+    /// @param[in] random_bins_com          a set of random (uniformly distributed) tresholds for each selected feature scaled at [0.0, 1.0]
+    /// @param[in] bin_offset_list          an array of offsets for each bin
+    /// @param[in] imp_data_list            an array of impurity data for each node
+    /// @param[in] nodeIndices              an array of node indices
+    /// @param[in] node_ind_ofs             a global offset for node indices
+    /// @param[in] node_list                a node structure containing split information
+    /// @param[in] left_child_imp_data_list an array of left child impurity values
+    /// @param[in] node_imp_dec_list        an array of node impurity decrease values
+    /// @param[in] update_imp_dec_required  boolean indicator to update impurity decrease structure
+    /// @param[in] node_count               a number of node to compute in current step
+    /// @param[in] deps                     a set of sycl event, which this method depends on
+    static sycl::event compute_random_split_by_histogram(
         sycl::queue& queue,
         const context_t& ctx,
-        const pr::ndarray<Bin, 2>& data,
-        const pr::ndview<Float, 1>& response,
-        const pr::ndarray<Index, 1>& tree_order,
+        const pr::ndarray<hist_type_t, 1>& node_hist_list,
         const pr::ndarray<Index, 1>& selected_ftr_list,
+        const pr::ndarray<Float, 1>& random_bins_com,
         const pr::ndarray<Index, 1>& bin_offset_list,
         const imp_data_t& imp_data_list,
-        const pr::ndarray<Index, 1>& node_ind_list,
+        const pr::ndarray<Index, 1>& nodeIndices,
```

Comment:
Why is this camelCase when everything else is snake_case?

Reply:
Old code fetching. Fixed.